web/src/components/organisms/Project/Schema/hooks.ts

```diff
@@ -0,0 +1,172 @@
+import { useCallback, useMemo, useState } from "react";
+
+import { Model } from "@reearth-cms/components/molecules/Schema/types";
+import {
+  useGetModelsQuery,
+  useCreateModelMutation,
+  useCreateFieldMutation,
+  SchemaFiledType,
+  SchemaFieldTypePropertyInput,
+} from "@reearth-cms/gql/graphql-client-api";
+
+type Params = {
+  projectId?: string;
+  modelId?: string;
+};
+
+export default ({ projectId, modelId }: Params) => {
+  const [modelModalShown, setModelModalShown] = useState(false);
+  const [fieldModalShown, setFieldModalShown] = useState(false);
+
+  const { data, refetch } = useGetModelsQuery({
+    variables: { projectId: projectId ?? "", first: 100 },
+    skip: !projectId,
+  });
+
+  const models = useMemo(() => {
+    return (data?.models.nodes ?? [])
+      .map<Model | undefined>(model =>
+        model
+          ? {
+              id: model.id,
+              description: model.description,
+              name: model.name,
+              key: model.key,
+              schema: {
+                id: model.schema.id,
+                fields: model.schema.fields.map(field => ({
+                  id: field.id,
+                  description: field.description,
+                  title: field.title,
+                  type: field.type,
+                  unique: field.unique,
+                  required: field.required,
+                })),
+              },
+            }
+          : undefined,
+      )
+      .filter((model): model is Model => !!model);
+  }, [data?.models.nodes]);
+
+  const rawModel = useMemo(
+    () => data?.models.nodes.find((p: any) => p?.id === modelId),
+    [data, modelId],
+  );
+  const model = useMemo(
+    () =>
+      rawModel?.id
+        ? {
+            id: rawModel.id,
+            description: rawModel.description,
+            name: rawModel.name,
+            key: rawModel.key,
+            schema: {
+              id: rawModel.schema.id,
+              fields: rawModel.schema.fields.map(field => ({
+                id: field.id,
+                description: field.description,
+                title: field.title,
+                type: field.type,
+                unique: field.unique,
+                required: field.required,
+              })),
+            },
+          }
+        : undefined,
+    [rawModel],
+  );
+
+  const [createNewModel] = useCreateModelMutation({
+    refetchQueries: ["GetModels"],
+  });
+
+  const handleModelCreate = useCallback(
+    async (data: { name: string; description: string; key: string }) => {
```

Comment:
For both this function and handleFieldCreate is there a good reason you are making the param an object? Since the mutations don't require an object isn't better to make the callback's params the same, then you can pass them in more simply

Reply:
@KaWaite , passing like this is much easier because of ant.d forms helpers, so I don't need to change on the component part.